agent/http.go

```diff
@@ -166,6 +167,12 @@ func (s *HTTPServer) handler(enableDebug bool) http.Handler {
 	return mux
 }
 
+// aclEndpointRE is used to find old ACL endpoints that take tokens in the URL
```

Comment:
I think the RE is too clever. I've read this a couple of times now and it isn't clear to me what it is trying to achieve. You could try to use multiple REs or at least provide some examples of what this is supposed to match. Since you're trying to obfuscate sensitive data I'd say clarity goes before efficiency. Maybe don't use an RE at all and work with url.Parse or strings.Split.